lib/index.js

```diff
@@ -101,3 +109,50 @@ exports.defaultDocumentLoader = defaultDocumentLoader;
 
 // expose for testing purposes
 exports._caches = require('./caches');
+
+/**
+ * Takes in a request to a kms system from a proxy and validates it.
+ *
+ * @param {object} options - Options to use.
+ * @param {Map<string, Array<string>>} options.allowedHosts - A Map
+ *   with a host and an optional Array of allowed ips.
+ * @param {string} options.host - An URL's host.
+ * @param {string} options.ip - An ip for the request.
+ *
+ * @throws {BedrockError} - A NotAllowedError if ip rules are broken.
+ *
+ * @returns {object} An object with the verify result and any errors.
+ */
+exports.validateRequest = ({allowedHosts, host, ip}) => {
+  let verified = false;
+  for(const [allowedHost, ips = []] of allowedHosts) {
+    // if we already found a valid host/ip entry stop
+    if(verified) {
+      break;
+    }
+    // if the host does not match the entry go to the next entry
+    if(host !== allowedHost) {
+      continue;
+    }
+    // if the entry matches the host, but there are no ip rules
+    // then verify and stop
+    if(!ips || ips.length === 0) {
+      verified = true;
+      break;
+    }
+    // if there are ip rules use them to set verified
+    verified = ips.includes(ip);
```

Comment:
This is the exact use case that `Map` aims to address.  See: https://medium.com/@prudywsh/reducing-time-complexity-using-a-map-15298ceeebcd

Perhaps there is some requirement that `ips` is modeled as an array.

Reply:
Using a Set for ips seems like a good idea.